code/Style.md

```diff
@@ -197,6 +241,24 @@ Cognitive load at 5 arguments is quite high already, so at that moment prefer co
 
 _Exclude `ctx` and `log` from the count of 5, since common arguments don't have as drastic effect._
 
+## Struct types constructors
+
+Constructor functions of struct types must always return a pointer to the type rather than a value.
```

Comment:
Why must they? It seems to me that there are several situations where it's better to return a value.

Also, implementing this rule would require a precise definition of "constructor function", and I'm not sure what that would look like.

Reply:
> Why must they? It seems to me that there are several situations where it's better to return a value.

I agree with that, I use a reference or a value depending of the case; however based on the response I wanted to know if for avoiding to think if a value fits better than a pointer, the convention is to always use a pointer, because a pointer works everywhere meanwhile a value not.

Another thing is about performance, which is always tricky to know without benchmarking/profiling and there isn't always time to spend for those, hence, I prefer to use a value or type based on semantics when there is not a clear gain and figuring out later on that there is a bottle neck and change it.

You know Go has transitioned from having performance issues with pointers over values but that started to change with the concurrent GC, so there isn't always a clear answer to use one or another without benchmarking/profiling.

> Also, implementing this rule would require a precise definition of "constructor function", and I'm not sure what that would look like.

It's a term used commonly in Go, mostly they are package exported functions which return an instance of a type and their names are commonly prefixed with `New`.

We can add that definition if this is confirmed as a convention and it gets added to the Style.

Reply:
Yeah, I would rather not have a convention around this. I generally agree with the statement, but might be over-arching.

I think the majority comes from the difference whether it's meant to be used as a value or as an object.

Reply:
Good to know that we don't want to have it as a convention, because I tended to use values or pointers depending of the semantics, however the answer that I've got in that comment, was that this probably was a convention.

I gonna remove this one.

Reply:
It's more because we implement services and similar things and they are usually pointers...